test/jdk/java/awt/im/4959409/bug4959409.java

```diff
@@ -21,52 +21,139 @@
  * questions.
  */
 
-/**
- *
+/*
+ * @test
  * @bug 4959409
- * @author Naoto Sato
+ * @summary Check whether pressing SHIFT + 1 triggers key event
+ * @key headful
  */
 
-import java.awt.*;
-import java.awt.event.*;
-import javax.swing.*;
+import java.awt.Container;
+import java.awt.AWTException;
+import java.awt.Robot;
+import java.awt.Rectangle;
+import java.awt.Point;
+import java.awt.BorderLayout;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyAdapter;
+import java.awt.event.InputEvent;
+import java.lang.reflect.InvocationTargetException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
 
-public class bug4959409 extends javax.swing.JApplet {
-    public void init() {
-        new TestFrame();
-    }
-}
+import javax.swing.JFrame;
+import javax.swing.JTextField;
+import javax.swing.JLabel;
+import javax.swing.SwingUtilities;
 
-class TestFrame extends JFrame implements KeyListener {
-    JTextField text;
-    JLabel label;
-
-    TestFrame () {
-        text = new JTextField();
-        text.addKeyListener(this);
-        label = new JLabel(" ");
-        Container c = getContentPane();
-        BorderLayout borderLayout1 = new BorderLayout();
-        c.setLayout(borderLayout1);
-        c.add(text, BorderLayout.CENTER);
-        c.add(label, BorderLayout.SOUTH);
-        setSize(300, 200);
-        setVisible(true);
-    }
+public class bug4959409 {
+
+    public final static int TIMEOUT = 20;
+    public final static int DELAY = 300;
+    private static JFrame frame;
+    private static JTextField jTextField;
+    private static JLabel jLabel;
+
+    public static void createUIAndTest() throws InterruptedException, InvocationTargetException, AWTException {
+        CountDownLatch frameVisibleLatch = new CountDownLatch(1);
+        CountDownLatch keyPressedEventLatch = new CountDownLatch(1);
+        final Point[] points = new Point[1];
+        final Rectangle[] rect = new Rectangle[1];
+
+        SwingUtilities.invokeAndWait(() -> {
+            frame = new JFrame("Test bug4959409");
+            jTextField = new JTextField();
+            jLabel = new JLabel();
+            frame.setLayout(new BorderLayout());
+            frame.addWindowListener(new WindowAdapter() {
+                @Override
+                public void windowOpened(WindowEvent e) {
+                    super.windowOpened(e);
+                    frameVisibleLatch.countDown();
+                    System.out.println("Frame opened");
+                }
+            });
+
+            jTextField.addKeyListener(new KeyAdapter() {
+
```

Comment:
I'd remove this blank line for consistency with the anonymous class above.

Reply:
Done